common.go

```diff
@@ -112,36 +113,66 @@ func validateBasicAuth(w http.ResponseWriter, r *http.Request, username string,
 	return true
 }
 
-// Regex for indentifying globbing patterns (or standard wildcards) in the whitelist.
+// Regex for indentifying globbing patterns (or standard wildcards) in the metrics allowlist and blocklist.
 var globbingPattern = regexp.MustCompile(`[|]|\*|\?|{|}|\\|!`)
 
-// Filter metrics based on configured whitelist.
-func getWhitelistedMetrics(rawMetrics map[string]metricType, whitelist []string, whitelistEnabled bool) map[string]metricType {
-	if whitelistEnabled {
-		whitelistedMetrics := make(map[string]metricType)
+// Filter metrics
+// Runs the raw metrics through allowlist first and the resulting metrics through blocklist
+func getFilteredMetrics(rawMetrics map[string]metricType, allowlist []string, allowlistEnabled bool, blocklist []string, blocklistEnabled bool) map[string]metricType {
+	filteredMetrics := filterAllowedMetrics(rawMetrics, allowlist, allowlistEnabled)
+	filterBlockedMetrics(filteredMetrics, blocklist, blocklistEnabled)
+
+	return filteredMetrics
+}
 
-		for _, stat := range whitelist {
+// Filter metrics based on configured allowlist.
+func filterAllowedMetrics(rawMetrics map[string]metricType, allowlist []string, allowlistEnabled bool) map[string]metricType {
+	if allowlistEnabled {
```

Comment:
This is not critical, but I usually prefer a short block return to long block continuum, to bring the complexity of the function down:
```go
if !allowlistEnabled {
   return rawMetrics
}
```